### DISCUSSION TITLE
How do I make conform.nvim use my .clang-format file?

### DISCUSSION THREAD

**Discussion by samuel-garmany** [LazyVim/LazyVim#2965] [Q&A]
I'm trying to make my c++ format according to the style in my `.clang-format` file that exists in the directory of my project. Unfortunately conform isn't using this file, but is formatting to clangd/llvm standards. How do I make my file take precedent?

I've added this to my lazy.lua but made no other changes:
```
{ import = "lazyvim.plugins.extras.lang.clangd" },
```

I'd assume I should install clang-format somewhere but the formatting already worked somehow and I'm not sure where to add the clang-format package.

**Comment by arslantu** [comment]
This may work for you.

Create `~/.config/nvim/lua/plugins/conform.lua` with content below:

```lua
return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        cpp = { "clang_format" },
      },
      formatters = {
        clang-format = {
          prepend_args = {"--style=file:<your_format_file_path>"},
        },
      },
    },
  },
}
```

**Reply by robertomorrison0** [reply]
```clang-format ``` should also be ```clang_format```

**Comment by sat0sh1c** [comment]
a little different variation of the variant above, so it will try to use the `.clang_format` file from the current directory first, and default to the one in nvim dir in the case of it being absent:
```lua
local function get_clang_format_path()
  local pwd = vim.fn.getcwd()
  local format_path
  local pwd_clang_format_hnd = io.open(".clang_format", "r")
  if pwd_clang_format_hnd == nil then
    format_path = os.getenv("HOME") .. "/.config/nvim/lua/config/.clang_format"
  else
    io.close(pwd_clang_format_hnd)
    format_path = pwd .. "/.clang_format"
  end
  return format_path
end

return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        cpp = { "clang_format" },
      },
      formatters = {
        clang_format = {
          prepend_args = { "--style=file:" .. get_clang_format_path() },
        },
      },
    },
  },
}
```
